src/app/pages/deleteContact/deleteContact.component.ts

```diff
@@ -36,14 +35,15 @@ export class DeleteContactComponent implements OnInit {
   constructor(
     private backend: BackendService,
     private router: Router,
-    private select: SelectService
+    private route: ActivatedRoute,
   ) {
-    this.cardId = this.select.deleteId
+    this.cardId = Number(this.route.snapshot.paramMap.get('id'));
   }
 
   ngOnInit() {
-    if (!this.select.deleteId) { return this.router.navigate(['allContacts']); }
-    this.backend.loadContact(this.select.deleteId)
+    let id = Number(this.route.snapshot.paramMap.get('id'));
```

Comment:
For these next two lines, you could just use `this.cardId` since it's already set in the constructor. Then you won't need to reference the route snapshot twice.